### PR TITLE
[doc] add tutorial.md

### PR DESCRIPTION
**Pull request by placebokkk**

### CODE REVIEW

File: docs/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
+## Tutorial
+
+If you meet any problems when going through this tutorial, please feel free to ask in github [issues](https://github.com/mobvoi/wenet/issues). And thanks for any kind feedbacks.
+
+### Setup environment
+- Clone
```
Comment (robin1001) [inline docs/tutorial.md:6]: 这一部分要不也同步到根目录下的README.md的Installation这个部分？

File: docs/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+#### stage 0 Prepare Kaldi format
+
+In this stage, `local/aishell_data_prep.sh` organizes the original aishell-1 data into a Kaldi format. In the kaldi dataset format, each dataset includes the two most important files `wav.scp`, `text`, and other resource files.
+
+In this demo recipe, this stage also uses Kaldi script `utils/perturb_data_dir_speed.sh` to do speed perturb augmentation on the training data. With 0.9 and 1.1 speeding up, the size of the training data is 3 times of the original data.
+
+please see `data/train/wav.scp` and `data/train/text`. 
+If you want to train your customized data, just organize the data into two files wav.scp and text, and start from `stage 1`.
+c
```
Comment (robin1001) [inline docs/tutorial.md:79]: remove this

File: docs/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+In this stage, Kaldi script `steps/make_fbank_pitch.sh` is used to extract the acoustic feature from the wavs for train, test, and dev set.
+
+`compute-cmvn-stats` is used to extract global cmvn statistics. These statistics will be used to normalize the acoustic feature. Set `cmvn=false` will ignore this step.
+
+See the generated training feature file in `fbank_pitch/train/feat.scp`.
+
+#### stage 2 Generate label token dictionary
+
+The dict, a map between label tokens(we use the character for Aishell-1) and
+ the integer indexes.
```
Comment (robin1001) [inline docs/tutorial.md:93]: <blank> denotes blank symbol for CTC.
<unk> denotes unknown token, any out of vocabulary token will be mapped into <unk>.
<sos/eos> denotes sos and eos symbol for attention encoder decoder training, and they shares the same id.

File: docs/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+#### stage 4 Neural Network training
+
+The NN model is training in this step.
+
+- multi-GPU mode
+
+If using DDP mode for multi-GPU, we suggest using `dist_backend="nccl"`. If the NCCL does not work, try using `gloo` or use `torch==1.6.0`
+Set the GPU ids in CUDA_VISIBLE_DEVICES. For example, set `export CUDA_VISIBLE_DEVICES="0,1,2,3,6,7"` to use card 0,1,2,3,6,7.
+
+- BreakPoint training
```
Comment (robin1001) [inline docs/tutorial.md:139]: How to resume training?

File: docs/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+- Decoding
+
+Recognition is also called decoding or inference. The function of the NN will be applied on the input acoustic feature sequence to out a text sequence.
+
+Four decoding methods are provided in wenet:
+
+* ctc_greedy_search: encoder + CTC greedy search
+* ctc_prefix_beam_search:  encoder + CTC prefix beam search
+* attention: encoder + attention-based decoder decoding
+* attention_rescoring: rescoring on the ctc candidates
```
Comment (robin1001) [inline docs/tutorial.md:175]: rescoring the ctc candidates from ctc prefix beam search with encoder output on attention-based decoder.

File: docs/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+
+In general, attention_rescoring is the best method. Please see [] for the details of these algorithms.
```
Comment (robin1001) [inline docs/tutorial.md:177]: 这里是准备以后加的吗？

File: docs/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+`--beam_size` is a tunable parameter, use a large beam size may get better results but also cause high computation cost.
+
+
```
Comment (robin1001) [inline docs/tutorial.md:181]: --batch_size can be greater than 1 for "ctc_greedy_search" and "attention" decoding mode, and it must be 1 for "ctc_prefix_beam_search" and "attention_rescoring" decoding mode.

**Review by robin1001 — CHANGES_REQUESTED** [review]
please see inline.

File: docs/tutorial.md (hunk 6 above)
Comment (placebokkk) [reply docs/tutorial.md]: 这里先放U2的论文链接了

File: docs/tutorial.md (hunk 1 above)
Comment (placebokkk) [reply docs/tutorial.md:6]: @whiteshirt0429 吴迪应该在改README，把这个也改一下。

Comment (robin1001) [reply docs/tutorial.md:6]: 这个我来加一下。

Comment (robin1001) [reply docs/tutorial.md:6]: 吴迪改的是aishell的README

**Review by robin1001 — APPROVED** [review]

**Pull request merged**